snapcraft/plugins/catkin.py

```diff
@@ -73,15 +73,36 @@ def schema(cls):
 
     def __init__(self, name, options):
         super().__init__(name, options)
+
         self.packages = set(options.catkin_packages)
-        self.dependencies = ['ros-core']
         self.package_deps_found = False
         self.package_local_deps = {}
         self._deb_packages = []
+        self.dependencies = []
 
-    def pull(self):
-        super().pull()
-        self._setup_deb_packages()
+        source_subdir = getattr(self.options, 'source_subdir', None)
```

Comment:
Ah, I probably stole it from a constructor somewhere, where it wasn't necessarily defined.